--- python3/Python/formatter_unicode.c	2019-10-01 19:23:17.000000000 +0200
+++ python3-android/Python/formatter_unicode.c	2019-10-16 23:35:42.049322393 +0200
@@ -689,6 +689,10 @@
 
 static const char no_grouping[1] = {CHAR_MAX};
 
+#if __ANDROID__ && (__ANDROID_API__ < 21)
+extern _Py_GetLocaleconvNumeric19(PyObject **decimal_point, PyObject **thousands_sep);
+#endif
+
 /* Find the decimal point character(s?), thousands_separator(s?), and
    grouping description, either for the current locale if type is
    LT_CURRENT_LOCALE, a hard-coded locale if LT_DEFAULT_LOCALE or
@@ -698,6 +702,15 @@
 {
     switch (type) {
     case LT_CURRENT_LOCALE: {
+#if __ANDROID__ && (__ANDROID_API__ < 21)
+         if (_Py_GetLocaleconvNumeric19(&locale_info->decimal_point,
+                                      &locale_info->thousands_sep) < 0) {
+             locale_info->grouping = no_grouping;
+             return -1;
+         }
+        locale_info->grouping = no_grouping;
+        break;
+#else
         const char *grouping;
         if (_Py_GetLocaleconvNumeric(&locale_info->decimal_point,
                                      &locale_info->thousands_sep,
@@ -715,6 +728,7 @@
         }
         locale_info->grouping = locale_info->grouping_buffer;
         break;
+#endif
     }
     case LT_DEFAULT_LOCALE:
     case LT_UNDERSCORE_LOCALE:
