--- python3/Python/pystrtod.c	2019-10-01 19:23:17.000000000 +0200
+++ python3-android/Python/pystrtod.c	2019-10-09 04:41:00.291481469 +0200
@@ -176,9 +176,14 @@
     assert(nptr != NULL);
 
     fail_pos = NULL;
-
-    locale_data = localeconv();
+//PMPP API<21
+#if __ANDROID_API__ < 21
+    decimal_point = ".";
+#else
+    locale_data = localeconv(); //PATCHED
     decimal_point = locale_data->decimal_point;
+#endif
+//PMPP API<21
     decimal_point_len = strlen(decimal_point);
 
     assert(decimal_point_len != 0);
