--- python3/setup.py	2019-10-01 19:23:17.000000000 +0200
+++ python3-android/setup.py	2019-10-07 18:21:21.624405264 +0200
@@ -655,6 +655,30 @@
             inc_dirs = (self.compiler.include_dirs +
                         sysroot_paths(('CPPFLAGS', 'CFLAGS', 'CC'),
                                       system_include_dirs))
+        #PMPP
+        print("===================================================================================================")
+        print("CROSS",cross_compiling)
+
+        for dir in ('/usr/include/x86_64-linux-gnu','/usr/local/include','/usr/include'):
+            if dir in inc_dirs:
+                inc_dirs.remove(dir)
+                print(' =========== REMOVED',dir)
+
+            if dir in self.compiler.include_dirs:
+                self.compiler.include_dirs.remove(dir)
+                print(' =========== REMOVED',dir)
+
+        print(self.compiler.include_dirs)
+        for dir in ('/usr/lib/x86_64-linux-gnu',):
+            if dir in system_lib_dirs:
+                system_lib_dirs.remove(dir)
+                print(' =========== REMOVED',dir)
+
+            if dir in self.compiler.library_dirs:
+                self.compiler.library_dirs.remove(dir)
+                print(' =========== REMOVED',dir)
+
+        print("===================================================================================================")
         exts = []
         missing = []
 
